builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,334 @@
+	tags := tagsFromMapRDS(d.Get("tags").(map[string]interface{}))
+
+	sourceIdsSet := d.Get("source_ids").(*schema.Set)
+	sourceIds := make([]*string, sourceIdsSet.Len())
+	for i, sourceId := range sourceIdsSet.List() {
+		sourceIds[i] = aws.String(sourceId.(string))
+	}
+
+	eventCategoriesSet := d.Get("event_categories").(*schema.Set)
+	eventCategories := make([]*string, eventCategoriesSet.Len())
+	for i, eventCategory := range eventCategoriesSet.List() {
+		eventCategories[i] = aws.String(eventCategory.(string))
+	}
+
+	request := &rds.CreateEventSubscriptionInput{
+		SubscriptionName: aws.String(name),
+		SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+		Enabled:          aws.Bool(d.Get("enabled").(bool)),
+		SourceIds:        sourceIds,
+		SourceType:       aws.String(d.Get("source_type").(string)),
+		EventCategories:  eventCategories,
+		Tags:             tags,
+	}
+
+	log.Println("[DEBUG] Create RDS Event Subscription:", request)
+
+	_, err := rdsconn.CreateEventSubscription(request)
+	if err != nil {
+		return fmt.Errorf("Error creating RDS Event Subscription %s: %s", name, err)
+	}
+
+	log.Println(
+		"[INFO] Waiting for RDS Event Subscription to be ready")
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating"},
+		Target:     []string{"active"},
+		Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta.(*AWSClient).rdsconn),
+		Timeout:    40 * time.Minute,
+		MinTimeout: 10 * time.Second,
+		Delay:      30 * time.Second, // Wait 30 secs before starting
+	}
+
+	// Wait, catching any errors
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Creating RDS Event Subscription %s failed: %s", d.Id(), err)
+	}
+
+	return resourceAwsDbEventSubscriptionRead(d, meta)
+}
+
+func resourceAwsDbEventSubscriptionRead(d *schema.ResourceData, meta interface{}) error {
+	sub, err := resourceAwsDbEventSubscriptionRetrieve(d.Get("name").(string), meta.(*AWSClient).rdsconn)
+	if err != nil {
+		return fmt.Errorf("Error retrieving RDS Event Subscription %s: %s", d.Id(), err)
+	}
+	if sub == nil {
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(*sub.CustSubscriptionId)
+	if err := d.Set("name", sub.CustSubscriptionId); err != nil {
+		return err
+	}
+	if err := d.Set("sns_topic", sub.SnsTopicArn); err != nil {
+		return err
+	}
+	if err := d.Set("source_type", sub.SourceType); err != nil {
+		return err
+	}
+	if err := d.Set("enabled", sub.Enabled); err != nil {
+		return err
+	}
+	if err := d.Set("source_ids", flattenStringList(sub.SourceIdsList)); err != nil {
+		return err
+	}
+	if err := d.Set("event_categories", flattenStringList(sub.EventCategoriesList)); err != nil {
+		return err
+	}
+	if err := d.Set("customer_aws_id", sub.CustomerAwsId); err != nil {
+		return err
+	}
+
+	// list tags for resource
+	// set tags
+	conn := meta.(*AWSClient).rdsconn
+	arn := buildRDSEventSubscriptionARN(d.Get("customer_aws_id").(string), d.Id(), meta.(*AWSClient).region)
+	d.Set("arn", arn)
```

Comment:
There is no `arn` field defined in the schema. I don't think it necessarily needs to be, but in such case we shouldn't be trying to save the ARN to state.